### ISSUE TITLE
Circular dependency with benchmarks?

### ISSUE THREAD

**Issue by yaxu** [tidalcycles/Tidal#536]
I'm struggling to build this with old-style cabal:
```
alex@stanage:~/src/tidal$ cabal configure --enable-benchmarks 
Warning: The configure command is a part of the legacy v1 style of cabal
usage.

Please switch to using either the new project style and the new-configure
command or the legacy v1-configure alias as new-style projects will become the
default in the next version of cabal-install. Please file a bug if you cannot
replicate a working v1- use case with the new-style commands.

For more information, see: https://wiki.haskell.org/Cabal/NewBuild

Resolving dependencies...
Warning: solver failed to find a solution:
Could not resolve dependencies:
[__0] trying: tidal-1.0.15 (user goal)
[__1] rejecting: tidal:!bench (constraint from config file, command line flag,
or user target requires opposite flag selection)
[__1] trying: tidal:*bench
[__2] unknown package: weigh (dependency of tidal *bench)
[__2] fail (backjumping, conflict set: tidal, weigh, tidal:bench)
After searching the rest of the dependency tree exhaustively, these were the
goals I've had most trouble fulfilling: tidal, tidal:bench, weigh
Trying configure anyway.
Configuring tidal-1.0.15...
cabal: Encountered missing dependencies:
criterion -any, weigh -any
```

I haven't looked closely, but this seems to be related to this long-standing bug: https://github.com/haskell/cabal/issues/1575

@nini-faroux any ideas?

It *does* work with v2-bench though, e.g. `cabal v2-bench bench-speed`

I didn't manage to work out how to add the `--time-limit=25` option though..

**Comment by nini-faroux** [comment]
oh, I didn't actually run into that issue. I used the usual cabal install, and then the 'cabal configure --enable-benchmarks' which you tried, and that worked ok for me. I think at one point, after adding the weigh library it asked me to reinstall criterion but other than that there was nothing strange. 

I just checked again now, and it's working ok for me, I attach a couple of screenshots of what I see when I run the benchmarks. 

![Screen Shot 2019-07-08 at 14 10 22](https://user-images.githubusercontent.com/31522663/60813800-5bd9bb00-a18c-11e9-9251-292b250446ae.png)

![Screen Shot 2019-07-08 at 14 12 28](https://user-images.githubusercontent.com/31522663/60813885-83c91e80-a18c-11e9-85b2-2a804c3b9c70.png)
